collectd-nginx-plus/README.md

```diff
@@ -0,0 +1,229 @@
+
+##### Built-in dashboards
+
+**Coming Soon**
+
+### Installation
+
+1. Download the [`niginx-plus-collectd`](https://github.com/signalfx/collectd-nginx-plus) project.
+
+1. Run the following command to install the module’s dependencies using `pip`, replacing the example path with the download location of the `niginx-plus-collectd` project:
+
+  ```
+  sudo pip install -r install_requirements.txt
+  ```
+
+1. Download SignalFx’s [sample configuration file](https://github.com/signalfx/integrations/blob/master/collectd-nginx-plus/10-nginx-plus.conf) for this plugin to `/etc/collectd/managed_config`.
+
+1. Modify the configuration file to provide values that make sense for your environment, as described in [Configuration](#configuration) below.
+
+1. Restart collectd.
+
+### Configuration
+
+Using the example configuration file [10-nginx-plus.conf](https://github.com/signalfx/integrations/tree/master/collectd-nginx-plus/10-nginx-plus.conf) as a guide, provide values for the configuration options listed below that make sense for your environment.
+
+| Configuration Option | Definition | Default Value |
+| ---------------------|------------|---------------|
+| ModulePath | Path on disk where collectd can find this module. | ``/usr/share/collectd/niginx-plus-collectd/plugin` |
+| StatusHost | IP address or DNS of the NGINX+ instance to retrieve status information from | `localhost` |
+| StatusPort | Port the NGINX+ status endpoint can be reached at. | `8080` |
+| DebugLogLevel | `true` to enable logging at DEBUG level. | `false` |
+| Username | Username to use for username/password authentication. | None |
+| Password | Password to use for username/password authentication. | None |
+| Dimension | A single additional dimension decorating to each metric. There are two values, the first for the name,
+the second for the value. | None |
+
+Example addition to the collectd configuration:
+
+```apache
+LoadPlugin python
+
+<Plugin python>
+  ModulePath "/usr/share/collectd/collectd-nginx-plus/plugin"
+  Import nginx_plus_collectd
+
+  <Module nginx_plus_collectd>
+    StatusHost "localhost"
+    StatusPort 8080
+    DebugLogLevel true
+    Username "user_1"
+    Password "my_password"
+    Dimension "extra_dimension_name_1" "extra_dimension_value_1"
+    Dimension "extra_dimension_name_2" "extra_dimension_value_2"
+  </Module>
+</Plugin>
+```
+
+By default only a small subset of the available metrics are published by default. The remaining metrics can be enabled by opting-in to additional metric groups. See [Metrics](#metrics) for more details on each metric group
+and how to enable them.
+
+### USAGE
+
+All metrics reported by the NGINX Plus collectd plugin will contain the following dimensions:
+
+* `nginx.version` will contain the version number of the NGINX Plus instance being monitored
+* `plugin` is always set to `nginx-plus`
+* `plugin_instance` will contain the IP address of the NGINX Plus instance as given in the `/status/address` response.
```

Comment:
@jay @mjtieman Is this plugin_instance unique enough?  Does /status/address give ip with port?  I could see this being a problem if a customer has more than one NGINX+ instance running on a host, but on different ports.